libraries/chain/include/eosio/chain/peer_keys_db.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <eosio/chain/controller.hpp>
+#include <boost/smart_ptr/atomic_shared_ptr.hpp>
+#include <boost/unordered/unordered_flat_map.hpp>
+
+namespace eosio::chain {
+
+/**
+ * This class caches the on-chain public keys that BP use to sign the `gossip_bp_peers`
+ * network message. These public keys are populated using the actions regpeerkey and
+ * delpeerkey of eos-system-contracts.
+ */
+class peer_keys_db_t {
+public:
+   using peer_key_map_t = boost::unordered_flat_map<name, public_key_type, std::hash<name>>;
+
+   peer_keys_db_t();
```

Comment:
This should be disabled by default. Most nodes have no use for this and there is no need to update it on those nodes. I was thinking that could be added later, but probably better to get that in before merging this.